From 43808dbb4d4ec948e58ff2820c13d4d1a93c6436 Mon Sep 17 00:00:00 2001
From: Volodymyr Pavlenko <pavlenko.vv@pm.me>
Date: Mon, 19 May 2025 10:29:24 +0300
Subject: [PATCH] bpl_cfg: DM: fix clients_measurement_mode

There is a logic bug, that returns false, tho ClientsMeasurementMode
is successfully read from DM. It should have correct logic to
fix the problem with MAP-4.7.4_ETH

Signed-off-by: Volodymyr Pavlenko <pavlenko.vv@pm.me>
---
 agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp | 4 +++-
 framework/platform/bpl/cfg/dm/bpl_cfg.cpp                | 6 ++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp b/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp
index ffb34152b..7bb0d8f9b 100644
--- a/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp
+++ b/agent/src/beerocks/fronthaul_manager/monitor/monitor.cpp
@@ -206,7 +206,7 @@ bool Monitor::thread_init()
         clients_measuremet_mode = bpl::eClientsMeasurementMode::ENABLE_ALL;
     }
     mon_db.set_clients_measuremet_mode(
-        (monitor_db::eClientsMeasurementMode)clients_measuremet_mode);
+        static_cast<monitor_db::eClientsMeasurementMode>(clients_measuremet_mode));
 #endif /* FEATURE_PRE_ASSOCIATION_STEERING */
 
     mon_db.set_radio_stats_enable(true);
@@ -916,6 +916,8 @@ bool Monitor::update_sta_stats(const std::chrono::steady_clock::time_point &time
             // This is a potentially long running operation.
             // If we haven't finished iterating on all stations, stop here and continue on next
             // method call from this point on.
+            LOG(WARNING)
+                << "update_sta_stats: long running operation, stop iterating on all stations";
             return true;
         }
 
diff --git a/framework/platform/bpl/cfg/dm/bpl_cfg.cpp b/framework/platform/bpl/cfg/dm/bpl_cfg.cpp
index be60d2fef..3849d6df8 100644
--- a/framework/platform/bpl/cfg/dm/bpl_cfg.cpp
+++ b/framework/platform/bpl/cfg/dm/bpl_cfg.cpp
@@ -198,10 +198,12 @@ int cfg_get_backhaul_params(int *max_vaps, int *network_enabled, int *preferred_
 
 bool cfg_get_clients_measurement_mode(eClientsMeasurementMode &clients_measurement_mode)
 {
-    int mode_value = 0;
-    if (read_agent_config_param("ClientsMeasurementMode", mode_value)) {
+    int mode_value = static_cast<int>(eClientsMeasurementMode::ENABLE_ALL);
+    if (!read_agent_config_param("ClientsMeasurementMode", mode_value)) {
+        MAPF_ERR("cfg_get_clients_measurement_mode: failed to read ClientsMeasurementMode\n");
         return false;
     }
+
     if (mode_value < static_cast<int>(eClientsMeasurementMode::DISABLE_ALL) ||
         mode_value >
             static_cast<int>(eClientsMeasurementMode::ONLY_CLIENTS_SELECTED_FOR_STEERING)) {
-- 
2.34.1

